app/models/king.rb

```diff
@@ -1,15 +1,98 @@
 class King < Piece
+
+  def y_position
```

Comment:
Ok - added it to enable the black king and black rook to castle. Have an issue that only lets white castle.